UnityProject/Assets/Scripts/Items/Wieldable.cs

```diff
@@ -0,0 +1,180 @@
+		}
+
+		public void OnInventoryMoveServer(InventoryMove info)
+		{
+			if (info.InventoryMoveType == InventoryMoveType.Remove)
+			{
+				isWielded = false;
+				ItemSlot hiddenHand = DetermineHiddenHand();
+
+				if (hiddenHand != null)
+				{
+					int hiddenHandSelection = (int) hiddenHandValues.bothHands;
+
+					if (hiddenHand.NamedSlot.GetValueOrDefault(NamedSlot.none) == NamedSlot.leftHand)
+					{
+						hiddenHandSelection = (int) hiddenHandValues.leftHand;
+					}
+					else if (hiddenHand.NamedSlot.GetValueOrDefault(NamedSlot.none) == NamedSlot.rightHand)
+					{
+						hiddenHandSelection = (int) hiddenHandValues.rightHand;
+					}
+
+					HideHand(PlayerManager.LocalPlayerScript.connectionToClient, false, hiddenHandSelection);
+				}
+			}
+		}
+
+		[TargetRpc]
+		private void HideHand(NetworkConnection target, bool HideState, int HiddenHandSelection)
+		{
+			HandsController.Instance.HideHands(HideState, HiddenHandSelection);
+		}
+
+		private ItemSlot DetermineHiddenHand()
+		{
+			ItemSlot hiddenHand = null;
+			var playerStorage = PlayerManager.LocalPlayerScript.DynamicItemStorage;
+			var currentSlot = playerStorage.GetActiveHandSlot();
+
+			var leftHands = playerStorage.GetNamedItemSlots(NamedSlot.leftHand);
+			foreach (var leftHand in leftHands)
+			{
+				if (leftHand != currentSlot && Validations.HasComponent<Wieldable>(leftHand.ItemObject) == false)
+				{
+					hiddenHand = leftHand;
+				}
+			}
+
+			var rightHands = playerStorage.GetNamedItemSlots(NamedSlot.rightHand);
+			foreach (var rightHand in rightHands)
+			{
+				if (rightHand != currentSlot && Validations.HasComponent<Wieldable>(rightHand.ItemObject) == false)
+				{
+					hiddenHand = rightHand;
+				}
+			}
+
+			return hiddenHand;
+		}
+
+		public bool WillInteract(HandActivate interaction, NetworkSide side)
+		{
+			if (!DefaultWillInteract.Default(interaction, side)) return false;
+
+			return true;
+		}
+
+		public void ServerPerformInteraction(HandActivate interaction)
+		{
+			ItemSlot hiddenHand = DetermineHiddenHand();
+
+			if (hiddenHand != null)
+			{
+				int hiddenHandSelection = (int) hiddenHandValues.bothHands;
+
+				if (hiddenHand.NamedSlot.GetValueOrDefault(NamedSlot.none) == NamedSlot.leftHand)
+				{
+					hiddenHandSelection = (int) hiddenHandValues.leftHand;
+				}
+				else if (hiddenHand.NamedSlot.GetValueOrDefault(NamedSlot.none) == NamedSlot.rightHand)
+				{
+					hiddenHandSelection = (int) hiddenHandValues.rightHand;
+				}
+
+				Inventory.ServerDrop(hiddenHand);
+
+				isWielded = !isWielded;
+
+				if (isWielded)
+				{
+					itemAttributes.ServerHitDamage = damageWielded;
+					itemAttributes.SetSprites(Wielded);
+					Chat.AddExamineMsgFromServer(interaction.Performer, $"You wield {gameObject.ExpensiveName()} grabbing it with both of your hands.");
+					HideHand(PlayerManager.LocalPlayerScript.connectionToClient, true, hiddenHandSelection);
+				}
+				else
+				{
+					itemAttributes.ServerHitDamage = damageUnwielded;
```

Comment:
need to apply this so, when it's removed from inventory it applies Correct damage, for example when the player gets Slipped or something like that